content/docs/design-principles.md

```diff
@@ -1,163 +1,163 @@
 
-React is not a generic data processing library. It is a library for building user interfaces. We think that it is uniquely positioned in an app to know which computations are relevant right now and which are not.
+React ümumi məlumat emalı kitabxanası deyil. Istifadəçi interfeysi yaratmaq üçün kitabxanadır. Düşünürük ki, applikasiya hansı hesablamaların aktual olduğunu və hansıların olmadığını bilməlidir.
 
-If something is offscreen, we can delay any logic related to it. If data is arriving faster than the frame rate, we can coalesce and batch updates. We can prioritize work coming from user interactions (such as an animation caused by a button click) over less important background work (such as rendering new content just loaded from the network) to avoid dropping frames.
+Bir şey ekrandan kənarda isə, bununla əlaqəli hər hansı bir məntiqi təxirə sala bilərik. Verilənlər çərçivələrin yenilənməsi vaxtından daha sürətli gəlsə, birləşdirmə və toplu yeniləmələr edə bilərik. Dropping frames-in qarşısını almaq üçün istifadəçi ilə qarşılıqlı əlaqədən yaranan işləri (məsələn, düymə basma animasiyasını) daha az əhəmiyyətli fon işlərindən üstün tuta bilərik (məsələn, şəbəkədən yüklənmiş komponenti göstərmədən).
 
-To be clear, we are not taking advantage of this right now. However the freedom to do something like this is why we prefer to have control over scheduling, and why `setState()` is asynchronous. Conceptually, we think of it as "scheduling an update".
+Bunun hələ həyata keçirilmədiyini bilin. Bununla birlikdə, bu sərbəstlik planlaşdırmaya niyə nəzarət etməyi seçdiyimizi və `setState()` funksiyasının niyə asinxron şəkildə işlədiyini göstərir. Konseptual olaraq, biz bunu "bir yeniləmənin planlaşdırılması" kimi düşünürük.
 
-The control over scheduling would be harder for us to gain if we let the user directly compose views with a "push" based paradigm common in some variations of [Functional Reactive Programming](https://en.wikipedia.org/wiki/Functional_reactive_programming). We want to own the "glue" code.
+İstifadəçilərə [Funksional Reaktiv Proqramlaşdırmada](https://en.wikipedia.org/wiki/Functional_reactive_programming) ümumi olan "push" paradiqmasının istifadəsinə icazə versəydik, planlaşdırmaya nəzarət etməyimiz daha çətin olardı. Kodumuzun "birləşdirən" olmasını istəyirik.
 
-It is a key goal for React that the amount of the user code that executes before yielding back into React is minimal. This ensures that React retains the capability to schedule and split work in chunks according to what it knows about the UI.
+React üçün əsas məqsəd React-ə qayıtmadan əvvəl yerinə yetirilən istifadəçi kodunun az olmasıdır. Bu, React-a UI haqqında bildiyinə görə, işləri planlaşdırma və bölmə qabiliyyətini verir.
```

Comment:
> Bu, React-a UI haqqında bildiyinə görə, işləri planlaşdırma və bölmə qabiliyyətini verir.

Bu, React-in UI haqqında **biliklərinə** əsasında işləri planlaşdırılması və bölünməsi qabiliyyətini verir.